proposals/mqtt.md

```diff
@@ -0,0 +1,83 @@
+# [MQTT](https://mqtt.org/) Discovery Handler
+
```

Comment:
Can we add a section on how the Akri broker will use the information provided? If i understand MQTT well enough, doesn't the Akri broker essentially just subscribe to the same MQTT broker as the Akri DH. To me, this makes "discovery" hard to understand, potentially similar to OPC with Akri. Might be worth adding a section that talks about how this is still useful.

Reply:
The Akri broker would indeed connect to the same MQTT broker.
It might however subscribe to different topics: for the pressure+temperature example, you could do discovery on temperature (so DH subscribe to `weather/+/temperature`) and have the Akri broker subscribe to all topics of the discovered instance (so subscribe to `weather/paris/#` for example).
The Akri broker would use the information provided to identify the topics to operate on and really identify the device, as the way the topics are crafted is not standardized, we can't infer a proper device identifier from it, so we just give the whole information, and let the Akri broker know better about what is the device identifier in the topic name.

About why this is needed, see a complex machine for example, you can spawn an application only able to handle one of them at a time and thus spawn it dynamically for every such machine plugged to your MQTT broker, this way you have the same overall configuration whether you have one machine or ten machines.
Similarly, as a manufacturer I could provide a standard set of Akri Configuration to spawn the needed user facing apps depending on what exact model or set of options you as an end-user enabled (so basically if machine talks to topic `/mymachine/extension/X` then spawn a pod that handles this specific extension, but if not then extension not enable and the pod is not needed).

I'm not sure this is really different from how the other DH are used :thinking: